subprojects/core/src/main/java/org/gradle/api/internal/project/LifecycleAwareProject.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.project;
+
+import org.gradle.internal.Cast;
+import org.gradle.internal.metaobject.DynamicInvokeResult;
+import org.gradle.internal.metaobject.DynamicObject;
+import org.gradle.internal.metaobject.DynamicObjectUtil;
+import org.gradle.internal.reflect.Instantiator;
+
+import javax.annotation.Nullable;
+
+
+public class LifecycleAwareProject extends MutableStateAccessAwareProject {
```

Comment:
💅  javadoc

Reply:
Added plain javadoc. Next PR will extend it

Reply:
Thanks! The formatting is a bit off, though